exchanges/lbank/lbank.go

```diff
@@ -0,0 +1,499 @@
+package lbank
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"github.com/thrasher-/gocryptotrader/common"
+	"github.com/thrasher-/gocryptotrader/config"
+	exchange "github.com/thrasher-/gocryptotrader/exchanges"
+	"github.com/thrasher-/gocryptotrader/exchanges/request"
+	"github.com/thrasher-/gocryptotrader/exchanges/ticker"
+	log "github.com/thrasher-/gocryptotrader/logger"
+)
+
+// Lbank is the overarching type across this package
+type Lbank struct {
+	exchange.Base
+	privateKey    *rsa.PrivateKey
+	privKeyLoaded bool
+	WebsocketConn *websocket.Conn
+	privKeyMutex  sync.Mutex
+}
+
+const (
+	lbankAPIURL     = "https://api.lbkex.com"
+	lbankAPIVersion = "1"
+	lbankRateLimit  = time.Second
+
+	// Public endpoints
+	lbankTicker         = "ticker.do"
+	lbankCurrencyPairs  = "currencyPairs.do"
+	lbankMarketDepths   = "depth.do"
+	lbankTrades         = "trades.do"
+	lbankKlines         = "kline.do"
+	lbankPairInfo       = "accuracy.do"
+	lbankUSD2CNYRate    = "usdToCny.do"
+	lbankWithdrawConfig = "withdrawConfigs.do"
+
+	// Authenticated endpoints
+	lbankUserInfo          = "user_info.do"
+	lbankPlaceOrder        = "create_order.do"
+	lbankCancelOrder       = "cancel_order.do"
+	lbankQueryOrder        = "orders_info.do"
+	lbankQueryHistoryOrder = "orders_info_history.do"
+	lbankOpeningOrders     = "orders_info_no_deal.do"
+	lbankWithdrawalRecords = "withdraws.do"
+	lbankWithdraw          = "withdraw.do"
+	lbankRevokeWithdraw    = "withdrawCancel.do"
+)
+
+// SetDefaults sets the basic defaults for Lbank
+func (l *Lbank) SetDefaults() {
+	l.Name = "Lbank"
+	l.RESTPollingDelay = 10
+	l.RequestCurrencyPairFormat.Delimiter = "_"
+	l.ConfigCurrencyPairFormat.Delimiter = "_"
+	l.AssetTypes = []string{ticker.Spot}
+	l.Requester = request.New(l.Name,
+		request.NewRateLimit(lbankRateLimit, 0),
```

Comment:
change lbankRateLimit -> time.Second; add secondary lbankAuthRateLimit && lbankUnauthRateLimit = 0 as a second parameter for this function and the one below.